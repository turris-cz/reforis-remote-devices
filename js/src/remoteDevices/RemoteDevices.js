/*
 * Copyright (C) 2019-2021 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";

import AddDeviceForm from "./AddDeviceForm";
import Devices from "./Devices";

RemoteDevices.propTypes = {
    ws: PropTypes.object.isRequired,
};

export default function RemoteDevices({ ws }) {
    return (
        <>
            <h1>{_("Devices")}</h1>
            <p>
                {_(
                    "You can set up other Turris routers to be managed remotely by this device."
                )}
            </p>
            <p>
                {_(
                    'To authorize your router, you will need a token file that can be obtained with the "Remote Access" plugin on the target device.'
                )}
            </p>
            <AddDeviceForm />
            <Devices ws={ws} />
        </>
    );
}
