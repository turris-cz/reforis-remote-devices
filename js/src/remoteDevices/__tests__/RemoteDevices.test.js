/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import mockAxios from "jest-mock-axios";
import { render, waitForElement, getByText } from "foris/testUtils/customTestRender";
import { WebSockets } from "foris";

import RemoteDevices from "../RemoteDevices";

describe("<RemoteDevices />", () => {
    it("should render component", async () => {
        const webSockets = new WebSockets();
        const { container } = render(<RemoteDevices ws={webSockets} />);
        expect(mockAxios.get).toBeCalledWith("/reforis/remote-devices/api/devices", expect.anything());
        mockAxios.mockResponse({ data: [] });
        await waitForElement(() => getByText(container, "No devices added yet."));
        expect(container).toMatchSnapshot();
    });
});
