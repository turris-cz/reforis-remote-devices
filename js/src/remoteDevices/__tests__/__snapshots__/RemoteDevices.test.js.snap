// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RemoteDevices /> should render component 1`] = `
<div>
  <h1>
    Devices
  </h1>
  <p>
    You can set up other Turris routers to be managed remotely by this device.
  </p>
  <p>
    To authorize your router, you will need a token file that can be obtained with the "Remote Access" plugin on the target device.
  </p>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
  >
    <h2>
      Add Device
    </h2>
    <p>
      Upload token file generated by the device you want to control remotely.
    </p>
    <form
      class="position-relative"
    >
      <div
        class="mb-3"
      >
        <label
          class="form-label custom-file-label"
          for="1"
        >
          Choose token file...
        </label>
        <div
          class="input-group custom-file"
        >
          <input
            accept=".tar.gz,.tgz"
            class="form-control custom-file-input"
            files=""
            id="1"
            type="file"
          />
        </div>
      </div>
      <div
        class="text-end"
      >
        <button
          class="btn btn-primary col-12 col-md-3 col-lg-2 d-inline-flex justify-content-center align-items-center"
          data-testid="upload-token-button"
          disabled=""
          type="submit"
        >
          <span>
            Upload token
          </span>
        </button>
      </div>
    </form>
  </div>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
  >
    <h2>
      Devices List
    </h2>
    <p
      class="text-muted text-center mb-0"
    >
      No devices added yet.
    </p>
  </div>
</div>
`;
